### PR TITLE
Refine some error msg, comment and document

### PR DESCRIPTION
**Pull request by wenyongh**

### CODE REVIEW

File: core/iwasm/interpreter/wasm_loader.c [hunk 1]
```diff
@@ -3503,7 +3503,7 @@ check_wasi_abi_compatibility(const WASMModule *module, bool main_module,
                 ->func_type;
         if (func_type->param_count || func_type->result_count) {
             set_error_buf(error_buf, error_buf_size,
-                          "The builtin _start() is with a wrong signature");
+                          "the signature of builtin _start() func is wrong");
```
Comment (lum1n0us) [inline core/iwasm/interpreter/wasm_loader.c:3506]: feel it is ok either "__start function" or "__start()". Since "()" means a function.

Comment (wenyongh) [reply core/iwasm/interpreter/wasm_loader.c]: Done

**Review by xujuntwt95329 — COMMENTED** [review]
LGTM

**Review by lum1n0us — APPROVED** [review]
LGTM

**Pull request merged**